MediaBrowser.Providers/Lyric/LrcLyricProvider.cs

```
@@ -0,0 +1,168 @@
+    /// <param name="item">The item to to process.</param>
+    /// <returns>If provider can determine lyrics, returns a <see cref="LyricResponse"/> with or without metadata; otherwise, null.</returns>
+    public LyricResponse? GetLyrics(BaseItem item)
+    {
+        string? lyricFilePath = LyricInfo.GetLyricFilePath(this, item.Path);
+
+        if (string.IsNullOrEmpty(lyricFilePath))
+        {
+            return null;
+        }
+
+        List<Controller.Lyrics.Lyric> lyricList = new List<Controller.Lyrics.Lyric>();
+        List<LrcParser.Model.Lyric> sortedLyricData = new List<LrcParser.Model.Lyric>();
+
+        IDictionary<string, string> fileMetaData = new Dictionary<string, string>();
+        string lrcFileContent = System.IO.File.ReadAllText(lyricFilePath);
+
+        try
+        {
+            // Parse and sort lyric rows
+            LyricParser lrcLyricParser = new LrcParser.Parser.Lrc.LrcParser();
+            Song lyricData = lrcLyricParser.Decode(lrcFileContent);
+            sortedLyricData = lyricData.Lyrics.Where(x => x.TimeTags.Count > 0).OrderBy(x => x.TimeTags.First().Value).ToList();
+
+            // Parse metadata rows
+            var metaDataRows = lyricData.Lyrics
+                .Where(x => x.TimeTags.Count == 0)
+                .Where(x => x.Text.StartsWith('[') && x.Text.EndsWith(']'))
+                .Select(x => x.Text)
+                .ToList();
+
+            foreach (string metaDataRow in metaDataRows)
+            {
+                var metaDataField = metaDataRow.Split(':');
+                if (metaDataField.Length != 2)
+                {
+                    continue;
+                }
+
+                string metaDataFieldName = metaDataField[0][1..].Trim().ToLowerInvariant();
+                string metaDataFieldValue = metaDataField[1][..^1].Trim();
+
+                fileMetaData.Add(metaDataFieldName, metaDataFieldValue);
+            }
+        }
+        catch
+        {
+            return null;
```

Would be nice to log so we know why the exception was thrown instead of just silently returning